block/block.md

```diff
@@ -1 +1,95 @@
 # Block
+
+## Abstract
+
+The block manager is a key component of full nodes (including sequencer) that is responsible for block production (in case of sequencer full node) or block syncing (syncing in this context includes retrieving the published blocks from the network, validating them to raise fraud proofs upon validation failure, and updating the state and storing the validated blocks). A full node invokes multiple of the block manager functionalities in parallel, such as:
+
+* block production (in case of sequencer full nodes)
+* block publication to DA network
+* block retrieval from DA network
+* block retrieval from Blockstore (which retrieves blocks from the p2p network)
+* block syncing
+
+## Protocol/Component Description
+
+The block manager is initialized using several parameters as defined below:
+
+* `signing key`: used for signing the blocks after production
+* `config`: block manager configurations
+* `genesis`: initialize the block manager with genesis state
+* `localstore`: store blocks and states
+* `(mempool, proxyapp, eventbus)`: for initializing the executor (state transition function). mempool is also used in the manager to check for availability of transactions for lazy block production
+* `dalc`: to submit and retrieve blocks to DA network  
```

Comment:
unless an abbreviation has previously been defined in the file, we should call out the full name in the description,
```suggestion
* `dalc`: the data availability light client used to submit and retrieve blocks to DA network  
```